furusiya/model/helper_functions/menu.py

```diff
@@ -68,4 +67,4 @@ def inventory_menu(header):
     # if an item was chosen, return it
     if index is None or len(Game.instance.inventory) == 0:
         return None
-    return Game.instance.inventory[index].get_component(Item)
+    return Game.instance.item_system.get(Game.instance.inventory[index])
```

Comment:
This makes me think three times. `Game.instance.inventory` used to be the player's inventory. Should we flag items in the `item_system` as being part of inventory? Probably not. So I think this is okay. But it looks, well, odd.

What is in `Game.instance.inventory`? is it a list of item components? Is that okay?

Reply:
It seems that `Game.instance.inventory` is a list of entities with the `Item` component.
`Game.instance.inventory` is still the player's inventory; but since items are technically components(for some reason!), that meant they needed to be managed.

They used to be managed by some component management methods of GameObject, but they were the only ones using it, so I substituted it for a component system.